compiler/noirc_frontend/src/elaborator/path_resolution.rs

```diff
@@ -102,12 +102,35 @@ enum MethodLookupResult {
     FoundMultipleTraitMethods(Vec<(TraitId, Ident)>),
 }
 
+/// Determines whether datatypes found along a path are to be marked as referenced
+/// or used (see UsageTracker::mark_as_referenced and UsageTracker::marke_as_used)
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(super) enum PathResolutionMode {
+    MarkAsReferenced,
+    MarkAsUsed,
+}
```

Comment:
Can you turn these references into links for the docs?

Reply:
Done!